noir-projects/aztec-nr/aztec/src/encrypted_logs/encrypted_note_emission.nr

```diff
@@ -66,9 +21,8 @@ fn emit_with_keys<Note, let N: u32, let NB: u32, let M: u32>(
     assert(note_exists, "Can only emit a note log for an existing note.");
```

Comment:
This sort of check, along with some of the things we did for e.g. pop notes and destroy note unsafe, make me think that we may want to e.g. use the type system to indicate whether a note has been read or not. So that if you're passing here a note that you got from `get_notes`, you'd not need to do this check.

Reply:
Good idea. It could be a simple wrapper around a generic Note. Shall we create an issue?

Reply:
Yes please, it'd be a bit of an experimental thing.

Reply:
https://github.com/AztecProtocol/aztec-packages/issues/8589